src/librustc_typeck/lib.rs

```diff
@@ -68,6 +68,7 @@ This API is completely unstable and subject to change.
 #![feature(rustc_diagnostic_macros)]
 #![feature(slice_patterns)]
 #![feature(never_type)]
+#![feature(inner_deref)]
```

Comment:
what's up with this?

Reply:
Used in some places, e.g. https://github.com/rust-lang/rust/pull/61857/files#diff-4dca14d9f8d48a6af9ed414a126a9823R243 (just search for `.deref()` in `expr.rs`) to make things type-check :)

Reply:
Is that an addition to `Option`? Damn, I think it's 1. confusing 2. we should be making progress on the coerce front instead of adding method hacks :/

Reply:
Yep; there's a PR to rename it to `.as_deref()`. It's much less of a hack than using `.map(...)`. =P

Reply:
We should probably never stabilize it and introduce the structural coercion already, hmpf.